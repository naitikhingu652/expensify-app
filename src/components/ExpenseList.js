import React from 'react';
import { connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem';
import selectexpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
  <div>
    {
      props.expenses.length === 0 ? (
        <p>No Expenses</p>
      ) : (
        props.expenses.map((expense) => <ExpenseListItem key={expense.id} {...expense} />)
      )
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectexpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
