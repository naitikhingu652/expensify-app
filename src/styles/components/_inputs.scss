.text-input {
  border: 1px solid #cacccd;
  height: 4.6rem;
  font-size: $font-size-large;
  font-weight: 300;
  padding: $s-size;
}

.select-input {
  @extend .text-input;
}

.textarea-input {
  @extend .text-input;
  height: 10rem;
}