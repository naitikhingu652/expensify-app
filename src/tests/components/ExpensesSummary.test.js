import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render ExpensesSumarry with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={250} />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSumarry with multiple expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={4} expensesTotal={25024} />);
  expect(wrapper).toMatchSnapshot();
});