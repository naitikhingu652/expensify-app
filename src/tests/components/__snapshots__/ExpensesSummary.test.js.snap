// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSumarry with multiple expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        4
      </span>
       
      expenses
       totalling 
      <span>
        $250.24
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render ExpensesSumarry with one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $2.50
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
